presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/IOPlanPrinter.java

```diff
@@ -0,0 +1,396 @@
+        private final List<FormattedRange> ranges;
+
+        @JsonCreator
+        public FormattedDomain(
+                @JsonProperty("nullsAllowed") boolean nullsAllowed,
+                @JsonProperty("ranges") List<FormattedRange> ranges)
+        {
+            this.nullsAllowed = nullsAllowed;
+            this.ranges = ImmutableList.copyOf(requireNonNull(ranges, "ranges is null"));
+        }
+
+        @JsonProperty
+        public boolean isNullsAllowed()
+        {
+            return nullsAllowed;
+        }
+
+        @JsonProperty
+        public List<FormattedRange> getRanges()
+        {
+            return ranges;
+        }
+    }
+
+    public static class FormattedRange
+    {
+        private final FormattedMarker low;
+        private final FormattedMarker high;
+
+        @JsonCreator
+        public FormattedRange(
+                @JsonProperty("low") FormattedMarker low,
+                @JsonProperty("high") FormattedMarker high)
+        {
+            this.low = requireNonNull(low, "low is null");
+            this.high = requireNonNull(high, "high is null");
+        }
+
+        @JsonProperty
+        public FormattedMarker getLow()
+        {
+            return low;
+        }
+
+        @JsonProperty
+        public FormattedMarker getHigh()
+        {
+            return high;
+        }
+    }
+
+    public static class FormattedMarker
+    {
+        private final Optional<String> value;
+        private final Bound bound;
+
+        @JsonCreator
+        public FormattedMarker(
+                @JsonProperty("value") Optional<String> value,
+                @JsonProperty("bound") Bound bound)
+        {
+            this.value = requireNonNull(value, "value is null");
+            this.bound = requireNonNull(bound, "bound is null");
+        }
+
+        @JsonProperty
+        public Optional<String> getValue()
+        {
+            return value;
+        }
+
+        @JsonProperty
+        public Bound getBound()
+        {
+            return bound;
+        }
+    }
+
+    private class IOPlanVisitor
+            extends PlanVisitor<Void, IOPlanBuilder>
+    {
+        @Override
+        protected Void visitPlan(PlanNode node, IOPlanBuilder context)
+        {
+            node.getSources().forEach(plan -> plan.accept(this, context));
+            return null;
+        }
+
+        @Override
+        public Void visitTableScan(TableScanNode node, IOPlanBuilder context)
+        {
+            TableMetadata tableMetadata = metadata.getTableMetadata(session, node.getTable());
+            context.addInputTableColumnInfo(new IOPlan.TableColumnInfo(
+                    new QualifiedTablePrefix(
+                            tableMetadata.getConnectorId().getCatalogName(),
+                            tableMetadata.getTable().getSchemaName(),
+                            tableMetadata.getTable().getTableName()),
+                    parseConstraints(node.getTable(), node.getCurrentConstraint())));
```

Comment:
Why is this type of EXPLAIN called "IO"? It doesn't seem to provide any I/O related information.

Reply:
IO here means input and output. This produces the input tables (with the constraints) and the output table.